app/assets/javascripts/filter_tag.js

```diff
@@ -1,28 +1,57 @@
-myApp.controller('ItemsController',
-  ['$scope', '$routeParams', '$http', function MenuCtrl($scope, $routeParams, $http) {
-    var params = $routeParams.restaurant_id;
-    $http.get('/api/restaurants/'+params+'/items').then(function successCallback(response) {
+myApp.controller('ItemsController', ['$scope', '$routeParams', '$http',
+  function MenuCtrl($scope, $routeParams, $http) {
+  var params = $routeParams.restaurant_id;
+  $http.get('/api/restaurants/'+params+'/items').then(function successCallback(response) {
     $scope.restaurantId = params;
     $scope.restaurant = response.data;
-
+    
     $scope.currentMenu = $scope.restaurant.menus[0];
-    $scope.menuItems = [];
-    $scope.findMenuItems();
-
+    $scope.menuItems = $scope.findMenuItems();
+    $scope.popularItems = $scope.findPopular();
+    
     $scope.tagType = ["diets", "tastes", "ingredients"];
     $scope.tagFilters = $scope.createTagFilters($scope.findUniqueTags());
-    },
-    function errorCallback(response) {
-      console.log("There was an error");
+  }, function errorCallback(response) {
+    console.log("There was an error");
   });
-
+  
   $scope.findMenuItems = function() {
+    items = [];
     for (var section in $scope.currentMenu.sections) {
       for (var item in $scope.currentMenu.sections[section].items) {
-        $scope.menuItems.push($scope.currentMenu.sections[section].items[item]);
+        items.push($scope.currentMenu.sections[section].items[item]);
       }
     }
-  }
+    return items;
+  };
+
+  $scope.findPopular = function() {
```

Comment:
Does this function get called when the page loads or when the trending button is clicked?